SeaPublicWebsite.Data/Migrations/20220708114101_UpdateHeatingTypeColumnInEpc.cs

```diff
@@ -0,0 +1,70 @@
+﻿using Microsoft.EntityFrameworkCore.Migrations;
+
+#nullable disable
+
+namespace SeaPublicWebsite.Data.Migrations
+{
+    public partial class UpdateHeatingTypeColumnInEpc : Migration
+    {
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+            migrationBuilder.AddColumn<int>(
+                name: "EpcHeatingType",
+                table: "Epc",
+                type: "integer",
+                nullable: true);
+            
+            // Convert the other heating type into the new column
+            // Original column enum had 8 options, so we stack these new values
+            // starting from 8
+            // We also ignore the 'other' option since it already exists in the normal heating type
+            migrationBuilder.Sql(@"
+                UPDATE public.""Epc""
+                SET ""EpcHeatingType"" = ""Epc"".""OtherHeatingType"" + 8
+                WHERE ""Epc"".""OtherHeatingType"" IS NOT NULL AND ""Epc"".""OtherHeatingType"" != 2");
+            
+            // Then copy over the regular heating type values
+            migrationBuilder.Sql(@"
+                UPDATE public.""Epc""
+                SET ""EpcHeatingType"" = ""Epc"".""HeatingType""
+                WHERE ""Epc"".""HeatingType"" IS NOT NULL");
+            
+            migrationBuilder.DropColumn(
+                name: "HeatingType",
+                table: "Epc");
+
+            migrationBuilder.DropColumn(
+                name: "OtherHeatingType",
+                table: "Epc");
+        }
+
+        protected override void Down(MigrationBuilder migrationBuilder)
+        {
+            migrationBuilder.AddColumn<int>(
+                name: "HeatingType",
+                table: "Epc",
+                type: "integer",
+                nullable: true);
+            
+            migrationBuilder.AddColumn<int>(
+                name: "OtherHeatingType",
+                table: "Epc",
+                type: "integer",
+                nullable: true);
+            
+            migrationBuilder.Sql(@"
+                UPDATE public.""Epc""
+                SET ""HeatingType"" = ""Epc"".""EpcHeatingType""
+                WHERE ""Epc"".""EpcHeatingType"" IS NOT NULL AND ""Epc"".""EpcHeatingType"" < 8");
+            
+            migrationBuilder.Sql(@"
+                UPDATE public.""Epc""
+                SET ""OtherHeatingType"" = ""Epc"".""EpcHeatingType"" - 8
+                WHERE ""Epc"".""EpcHeatingType"" IS NOT NULL AND ""Epc"".""EpcHeatingType"" >= 8");
+            
```

Comment:
I had more thought about this because it made me wonder....
Assuming the special case where we need to revert an 'other' answer into the `HeatingType` and `OtherHeatingType` columns, does that imply that the previous answers were Other and Other? If so (which I now think it's the only way to get an other response), I should add a separate sql update in the lines of `if EpcHeatingType == 7(Other) set OtherHeatingType = 2(other)`

Reply:
I think it's `if (EpcHeatingType == 7(Other) AND OtherHeatingType IS NULL) set OtherHeatingType = 2(other)`

Reply:
Ah correct! I'll update this once I delete the migration and redo it after the merge.